python/ray/rllib/evaluation/sampler.py

```diff
@@ -422,32 +422,38 @@ def _process_observations(base_env, policies, batch_builder_pool,
                     "episode": episode
                 })
             del active_episodes[env_id]
-            resetted_obs = base_env.try_reset(env_id)
-            if resetted_obs is None:
-                # Reset not supported, drop this env from the ready list
-                if horizon != float("inf"):
-                    raise ValueError(
-                        "Setting episode horizon requires reset() support "
-                        "from the environment.")
+
+            if hasattr(base_env, 'send_reset'):
```

Comment:
In the interest of avoiding making this (already long) function even longer, how about we keep the `try_reset` code path, but check `if resetted_obs == ASYNC_RESET`.

This would also avoid needing to add a new send_reset method.

Reply:
I agree that seems like a less of a hack. :) Did so.